lib/azure/armrest/armrest_collection.rb

```diff
@@ -4,6 +4,7 @@ module Azure
   module Armrest
     class ArmrestCollection < Array
       attr_accessor :continuation_token
+      attr_accessor :response_headers
```

Comment:
Add an initializer to accept the response_headers as an optional hash, and covert to model object internally. Then the calling can be greatly simplified.